src/infinitemps.jl

```diff
@@ -74,38 +84,57 @@ ITensors.findsites(ψ::InfiniteCanonicalMPS, T::ITensor) = findsites(ψ.AL, T)
 # For now, only represents nearest neighbor interactions
 # on a linear chain
 struct InfiniteITensorSum
```

Comment:
Maybe we should rename this to `InfiniteSum{T}`, then define `InfiniteITensorSum = InfiniteSum{ITensor}` and `InfiniteMPOSum = InfiniteSum{MPO}`?

Reply:
Then most code could be written in terms of `InfiniteSum`, for example, and certain methods could be specialized to the ITensor or MPO version.

Reply:
I think it is indeed the best way to do it. InfiniteSum{ } can be done easily and the only changes will be in contractions.

Reply:
Do you want me to have an attempt at it?

Reply:
Yeah, I could do that.